### PR TITLE
enable Geode and avatarite drops

### PR DESCRIPTION
**Pull request by TeoTwawki**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

This was made waaaaaay back in dakstar in early 2017. I hate it coz its tall n'ugly but it works about as well as our seal drops as far as retail accuracy. I am updating it to fit the modern codebase and styling.

### CODE REVIEW

**Comment by TeoTwawki** [comment]
its gonna be awhile, lots on my plate atm.

File: src/map/entities/mobentity.cpp [hunk 1]
```diff
@@ -1046,9 +1046,214 @@ void CMobEntity::DropItems(CCharEntity* PChar)
...
+                    case 7: // Dark
+                        if (dayRoll < 45)
+                            AddItemToPool(3527, ++dropCount) // Fenrite
+                        else
+                            AddItemToPool(3304, ++dropCount) // Shadow Geode
+                        break;
+                }
+            }
+        }
+        // Geodes. Wiki's have conflicting info on mob lv required. One says 50 the other 75. Betting 50 is correct since the other tier is 80.
```
Comment (kaincenteno) [inline src/map/entities/mobentity.cpp]: I agree with this, most likely it is mob level 50. Maybe add a TODO to easy find this comment whenever someone wants to volunteer some time and confirm this? :smile_cat:

Comment (RAIST5150) [reply src/map/entities/mobentity.cpp]: > 
> 
> I agree with this, most likely it is mob level 50. Maybe add a TODO to easy find this comment whenever someone wants to volunteer some time and confirm this? 😸

Seems like I used to get them off the Funguar and Treants and not just the Hippogryphs and such that can spawn at 75 in Grauberg (S) when I was running trials there.  Was a good starting zone for several elemental weapons, with most targets being in the 60's.  IDK about 50 being the lower boundary, but could see it being 60 0r 65.

Grr... could have sworn I recalled SE mentioning them somewhere ages ago. So far all I managed to dig up is when they added geodes to cactus stem results in Dec. of 2012.

Comment (TeoTwawki) [reply src/map/entities/mobentity.cpp]: > Seems like I used to get them off the Funguar and Treants and not just the Hippogryphs and such that can spawn at 75 in Grauberg (S) when I was running trials there. Was a good starting zone for several elemental weapons, with most targets being in the 60's. IDK about 50 being the lower boundary, but could see it being 60 0r 65.

@RAIST5150 my own personal theory is it will line up with the same lv mobs will be IEP to a lv99 when put to the test, which puts it lower than 75 but a little ways above the 50 ffxiclopedia claims. I recall getting some off exp mobs (IT's) while my character was on a job in the mid 50's back when they came out so I'm confident its at least lower than 75

Comment (RAIST5150) [reply src/map/entities/mobentity.cpp]: hehe... got lost in forums trying to see if I could spot something more concrete. Lowest levels I saw specifically cited fairly consistently were mobs that spawned around 60 (like a 58-62 range).  Some citings of mobs that could spawn lower, but some were not granting exp for them and they weren't sure of the levels that actually dropped the rocks, so it could actually be mobs that dip into the 50's.  Myself, I just remember targeting things that spawned around 60+ (like the Brasscaps and Knotty Treants I mentioned in Grauberg... 59-64 range).

It seemed to often beg the question of it basically being a mob that would grant exp to a 75 player.  IDK if that would be an acceptable rule of thumb to go with or not though.

Guess it is gonna be a hard one to figure out for certain at this point, with everyone being 99... forget how low the mobs can go and still grant EXP at 99.  Seems to be the only method one could use to figure out the mobs actual level?

Comment (TeoTwawki) [reply src/map/entities/mobentity.cpp]: if looking at it from the player side there quite a range of levels mobs will still con EP at 75. iirc the exp tables for that correctly, that would put the lowest lv mob to give exp at lv 55 or 56. At 99 the lowest mob to potentially give exp is 59 due to the introduction of IEP, so pretty close ballpark.

I was figuring on leaving this a draft till I have time to make sure and then correct it, rather than leaving those comments from 5 years ago. I still have some jobs at lv 50 I can use for this on retail. I likely won't get aroudn to it till mid November or so tho. in the meantime I will just be making sure the logic still works after having updated 5yr old code. (almost 6yrs actually)

**Comment by TeoTwawki** [comment]
thanks to @Wiggo32 on the capture discord all I need to confirm from retail now are the level thresholds
edit: got'em

File: src/map/entities/mobentity.cpp (hunk 1 above)
Comment (TeoTwawki) [reply src/map/entities/mobentity.cpp]: got confirmation it really is mob lv 50 or above. Also thought of a way to shorten the code by many lines, will update the branch when I have time and get this done.

Comment (kaincenteno) [reply src/map/entities/mobentity.cpp]: awesomeness great work! :D

**Comment by TeoTwawki** [comment]
Questions unanswered:
- What even are the drop rates supposed to be?
- Do these drop per kill, or per party member like crystals?
- IF so, can A geode and an Avatarite drop in the same kill?

For now, I have used a 20% rate on both on a per-kill basis however Avatarite won't drop if a geode does, which reduces it somewhat. This anecdotally "feels like retail" to me, but I have _nothing_ to base that on. There just isn't any data on drop rate that I was able to find back when I started this branch in 2017 DSP. [BG still has not much to go off now](https://www.bg-wiki.com/ffxi/Category:Avatarite) and 'clopedia [straight up pulls provably untrue nonsense out of its rear end](https://ffxiclopedia.fandom.com/wiki/Category:Geodes) - there is no 5 min wait between drops on these. I just got 3 back to back on retail last night in less than 2 min. ***EDIT:*** possibly the seals campaign currently going on affects geodes as well (as an unstated bonus).

**Comment by TeoTwawki** [comment]
tl;dr about as retail accurate as our current beastmen seal drop system. 

Logic and rates can be easily adjusted in the future when more detailed info emerges

**Review by claywar — APPROVED** [review]
